libnetwork/endpoint.go

```diff
@@ -64,9 +67,8 @@ func (ep *Endpoint) MarshalJSON() ([]byte, error) {
 		epMap["generic"] = ep.generic
 	}
 	epMap["sandbox"] = ep.sandboxID
-	epMap["anonymous"] = ep.anonymous
+	epMap["dnsNames"] = ep.dnsNames
 	epMap["disableResolution"] = ep.disableResolution
-	epMap["myAliases"] = ep.myAliases
```

Comment:
I wonder if we should keep populating `anonymous` and `myAliases` in the marshaled objects so users could downgrade. Though only live-restored containers would be affected given that it's the endpoint object. As we already document that [live-restore is only supported across patch upgrades](https://docs.docker.com/config/containers/live-restore/#live-restore-during-upgrades) we have no obligation to support major-version _downgrades_.

Reply:
Since we explicitly specify we don't support live-restore for major upgrades, I think we shouldn't care much about live-restore for major downgrades.